alerting/src/main/kotlin/org/opensearch/alerting/action/GetRemoteIndexesRequest.kt

```diff
@@ -36,7 +37,46 @@ class GetRemoteIndexesRequest : ActionRequest {
         out.writeBoolean(includeMappings)
     }
 
+    /**
+     * Validates the request [indexes].
+     * @return TRUE if all entries are valid; else FALSE.
+     */
+    fun isValid(): Boolean {
+        return indexes.isNotEmpty() && indexes.all { validPattern(it) }
+    }
+
+    /**
+     * Validates individual entries in the request [indexes].
+     *
+     * @param pattern The entry to evaluate. The expected patterns are `<index-pattern>` for a local index, and
+     * `<cluster-pattern>:<index-pattern>` for remote indexes. These patterns are consistent with the `GET _resolve/index` API.
+     * @return TRUE if the entry is valid; else FALSE.
+     */
+    private fun validPattern(pattern: String): Boolean {
+        // In some situations, `<cluster-pattern>` could contain a `:` character.
+        // Identifying the `<index-pattern>` based on the last occurrence of `:` in the pattern.
+        val separatorIndex = pattern.lastIndexOf(":")
+        return if (separatorIndex == -1) {
+            // Indicates a local index pattern.
+            INDEX_PATTERN_REGEX.matches(pattern)
+        } else {
+            // Indicates a remote index pattern.
+            val clusterPattern = pattern.substring(0, separatorIndex)
+            val indexPattern = pattern.substring(separatorIndex + 1)
+            CLUSTER_PATTERN_REGEX.matches(clusterPattern) && INDEX_PATTERN_REGEX.matches(indexPattern)
+        }
+    }
+
     companion object {
+        /**
+         * This regex asserts that the string:
+         *  Starts with a lowercase letter, digit, or asterisk
+         *  Contains a sequence of characters followed by an optional colon and another sequence of characters
+         *  The sequences of characters can include lowercase letters, uppercase letters, digits, underscores, asterisks, or hyphens
+         *  The total length of the string can range from 1 to 255 characters
+         */
+        val CLUSTER_PATTERN_REGEX = Regex("^(?=.{1,255}$)[a-z0-9*]([a-zA-Z0-9_*-]*:?[a-zA-Z0-9_*-]*)$")
```

Comment:
Let's expose this in common utils instead of defining it again here. Joanne did something similar in this PR: https://github.com/opensearch-project/common-utils/pull/630/files#diff-bef5bf4845e900a27dee298f88ab24e497693cd743cf34d97fb13c1fcaee0c95

Reply:
Good point. I'll revise this PR, and PR https://github.com/opensearch-project/common-utils/pull/633.